### ISSUE TITLE
False positive ST11 (possibly by `exclude (column)`) 

### ISSUE THREAD

**Issue by phammer** [sqlfluff/sqlfluff#6582]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

```sql
with
volume_blocks_with_company_ref_id as (
    select
        volume_blocks.*,
        contracts.* exclude (contract_id)
    from volume_blocks
    left join contracts on
        volume_blocks.contract_id = contracts.contract_id
)
```

triggers 

```plain
L:  34 | P:  15 | ST11 | Joined table 'contracts' not referenced elsewhere in                                                                                                                                                                                                                                              
                       | query [structure.unused_join]
```

which is not correct. `contracts` is referenced in the `select` statement.

**Remark:** there are 2 similar issues related to ST11. The reported SQLs on those issues looked different to mine, thus I opted to create a dedicated ticket. You might merge them again later.

### Expected Behaviour

Sqlfluff shouldn't report a violation of ST11 in this case

### Observed Behaviour

See above.

### How to reproduce

**SQL**

See issue description.

**Command**

`sqlfluff lint` will reproduce the issue as mentioned above.



### Dialect

snowflake

### Version

3.3.0

### Configuration

```
[sqlfluff]
templater = dbt
dialect = snowflake
exclude_rules = LT05, ST06, LT01

[sqlfluff:rules:references.keywords]
ignore_words = first_name,last_name,email,roles,language
```

```
target/
# dbt <1.0.0
dbt_modules/
# dbt >=1.0.0
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
Closing this issue as a duplicate of #6511 as this scenario is covered there and the fix for #6572 should handle this.
